src/lib.rs

```diff
@@ -283,6 +286,47 @@ impl<C> PipelineBuilder<C> {
     ///     .install_simple();
     /// ```
     pub fn with_trace_config(self, config: sdk::trace::Config) -> Self {
+        let merged_resource = match self.config {
+            Some(base_config) => base_config.resource.merge(&config.resource),
+            None => (*config.resource).clone(),
+        };
+
+        let config = config.with_resource(merged_resource);
+
+        PipelineBuilder {
+            config: Some(config),
+            ..self
+        }
+    }
+
+    /// Assign the service name under which to group traces by adding a service.name
+    /// `sdk::Resource` or overriding a previous setting of it.
+    ///
+    /// If a `sdk::Config` does not exist on the `PipelineBuilder` one will be created.
+    ///
+    /// This will be translated, along with the service namespace, to the Cloud Role Name.
+    ///
+    /// ```
+    /// # use opentelemetry::{KeyValue, sdk};
+    /// let tracer = opentelemetry_application_insights::new_pipeline("...".into())
+    ///     .with_client(reqwest::blocking::Client::new())
+    ///     .with_service_name("my-application")
+    ///     .install_simple();
+    /// ```
+    pub fn with_service_name<T: Into<String>>(self, name: T) -> Self {
+        let config = match self.config {
+            Some(config) => config,
+            None => sdk::trace::Config::default(),
+        };
```

Comment:
This can be shortened to:
```suggestion
        let config = self.config.unwrap_or_default();
```